### PR TITLE
stm32: drivers: flash: mcumgr: support for H7

### PR DESCRIPTION
**Pull request by JvanDooren**
Add support for setting flash word alignment size.
Update to tinycbor 0.5.4 for mcumgr (https://github.com/fancom/tinycbor/pull/1).

Signed-off-by: Jeroen van Dooren <jeroen.van.dooren@nobleo.nl>

### CODE REVIEW

File: drivers/flash/flash_stm32_qspi.c [hunk 1]
```diff
@@ -1094,6 +1094,9 @@ static int flash_stm32_qspi_init(const struct device *dev)
 
 	LOG_DBG("Device %s initialized", DEV_NAME(dev));
 
+	/* wait until quad output is set before doing an immediate read/write call */
+	qspi_wait_until_ready(dev);
```
Comment (heinwessels) [inline drivers/flash/flash_stm32_qspi.c:1096]: A little nit-picky, but this should be before `LOG_DBG("Device %s initialized", DEV_NAME(dev));`. Then it won't say it's initialized if the Flash never gets `ready` for some reason.

File: west.yml [hunk 2]
```diff
@@ -119,7 +125,9 @@ manifest:
       path: modules/audio/sof
     - name: tinycbor
       path: modules/lib/tinycbor
-      revision: 40daca97b478989884bffb5226e9ab73ca54b8c4
+      repo-path: tinycbor
+      remote: optima
+      revision: 69e745dd9d4975a770ca70195f61510b43e125d4
```
Comment (heinwessels) [inline west.yml]: This is not the final hash, the pull request on tinycbor should be merged first.

**Review by heinwessels — CHANGES_REQUESTED** [review]

File: drivers/flash/flash_stm32_qspi.c (hunk 1 above)
Comment (JvanDooren) [reply drivers/flash/flash_stm32_qspi.c:1096]: Will do

File: west.yml (hunk 2 above)
Comment (JvanDooren) [reply west.yml]: Will do

**Review by heinwessels — APPROVED** [review]

**Pull request merged**